packages/widgets/src/blocks/widget-box/edit.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalUseInnerBlocksProps as useInnerBlocksProps,
+	useBlockProps,
+	InnerBlocks,
+	RichText,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { createBlock } from '@wordpress/blocks';
+
+export default function Edit( { attributes, setAttributes, clientId } ) {
+	const innerBlocksProps = useInnerBlocksProps(
+		{
+			className: 'wp-widget-box__inner-blocks',
+		},
+		{
+			renderAppender: InnerBlocks.ButtonBlockAppender,
+		}
+	);
+
+	const blockProps = useBlockProps();
+
+	return (
+		<div { ...blockProps }>
+			<RichText
+				className="widget-title"
+				tagName="h2"
+				aria-label={ __( 'Widget title' ) }
+				placeholder={ __( 'Add a Widget title' ) }
+				value={ attributes.widgetTitle }
+				onChange={ ( value ) =>
+					setAttributes( { widgetTitle: value } )
+				}
+				onSplit={ ( value, isOriginal ) => {
+					let block;
+
+					if ( isOriginal || value ) {
+						block = createBlock( 'core/heading', {
+							...attributes,
```

Comment:
What other attributes do we want to spread here? If none then I think we can memoize this function (if necessary).

Reply:
Yeh this whole RichText area is subject to change. I kind-a copy/pasted from `core/heading` at this point. Needs refinement.